beaker-ts/src/notebook.ts

```diff
@@ -558,43 +586,118 @@ export class BeakerQueryCell extends BeakerBaseCell implements IQueryCell {
         this._current_input_request_message = undefined;
     }
 
-    public toMarkdownCell() {
-        const renderedMarkdownLines = [`# ${this.source}\n`];
-        this.events.forEach((event) => {
+    public toMultipleCells(): [BeakerMarkdownCell, ...BeakerBaseCell[]] {
+        // break apart a query cell into a series of cells so that code is interspersed with markdown cells
+        // and that in a jupyter notebook, loading and saving handles the redundant cells / query cell mapping
```

Comment:
The way the logic is implemented in this method is a bit confusing and difficult to follow. Perhaps we could pair program to clean this up as an exercise.